File: scratch/debug_stock.js
const { createClient } = require('@supabase/supabase-js');

const supabase = createClient('https://ztlrayekobgcllnxmqft.supabase.co', 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Inp0bHJheWVrb2JnY2xsbnhtcWZ0Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3ODgzNzc4NTIsImV4cCI6MjEwMzk1Mzg1Mn0.SCv_r5KOQIN0RTvEEQrZLCOGaaneWsPlJuIMnyxYXkE');

async function debugStock(category, brand, variant, qty) {
    console.log(`Checking stock for: [${category}] [${brand}] [${variant}] qty:${qty}`);
    
    const { data: inventory, error } = await supabase.from('inventory').select('*');
    if (error) {
        console.error("Fetch error:", error);
        return;
    }
    
    // Find matching items (maybe there are duplicates?)
    const matches = inventory.filter(i => i.category === category && i.brand === brand && i.variant === variant);
    console.log(`Found ${matches.length} matching items in database.`);
    
    if (matches.length > 0) {
        for (let i = 0; i < matches.length; i++) {
            const item = matches[i];
            console.log(`Match ${i+1}: ID=${item.id}, quantity=${item.quantity} (type: ${typeof item.quantity})`);
            console.log(`item.quantity >= qty is: ${item.quantity >= qty}`);
        }
    } else {
        console.log("No item found! Let's print all items with this brand just in case:");
        const brandMatches = inventory.filter(i => i.brand && i.brand.includes("GM Cuba"));
        console.log(brandMatches);
    }
}

// We don't know the exact category, but usually "Modular Plate" or "Switch"
debugStock("Modular Plate", "GM Cuba Series", "1M", 1);
debugStock("Switch", "GM Cuba Series", "1M", 1);
debugStock("Switch", "GM Cuba Series", "6A", 1);

File: scratch/debug_stock.py
import urllib.request
import json

SUPABASE_URL = 'https://ztlrayekobgcllnxmqft.supabase.co/rest/v1/inventory?select=*'
SUPABASE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Inp0bHJheWVrb2JnY2xsbnhtcWZ0Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3ODgzNzc4NTIsImV4cCI6MjEwMzk1Mzg1Mn0.SCv_r5KOQIN0RTvEEQrZLCOGaaneWsPlJuIMnyxYXkE'

headers = {
    'apikey': SUPABASE_KEY,
    'Authorization': f'Bearer {SUPABASE_KEY}',
    'Accept': 'application/json'
}

req = urllib.request.Request(SUPABASE_URL, headers=headers, method='GET')
try:
    with urllib.request.urlopen(req) as response:
        data = json.loads(response.read().decode('utf-8'))
        
        print(f"Total items in inventory: {len(data)}")
        
        print("\n--- Looking for GM Cuba Series ---")
        for item in data:
            if item.get('brand') and 'GM Cuba' in item.get('brand'):
                print(item)
                
        # Are there any items where variant has a trailing space?
        print("\n--- Looking for trailing spaces in variant ---")
        for item in data:
            v = item.get('variant', '')
            if v and (v.startswith(' ') or v.endswith(' ')):
                print("Trailing space found:", repr(v), "in item ID:", item.get('id'))
                
except Exception as e:
    print("Error:", e)
